### PR TITLE
feat: ListView - ui plugins

### LINKED ISSUE #366: ui.list_view - basic items + table support
Implement ui.list_view basic items + table support

### PR DESCRIPTION
**Pull request by bmingles**
ui.list_view - basic items + table support

resolves #366

### CODE REVIEW

**Review by mofojed — CHANGES_REQUESTED** [review]
Need to add some examples in the examples/README.md

I tried a couple of the examples as specified in the spec, and they didn't work as I was expecting. e.g. I can't change the selection here:
```
from deephaven import new_table
from deephaven.column import string_col, int_col

color_table = new_table([
    string_col("Keys", ["salmon", "lemonchiffon", "black"]),
    string_col("Labels", ["Salmon", "Lemon Chiffon", "Black"]),
    string_col("Descriptions", ["An interesting color", "Another interesting color", "A color"]),
    string_col("Icons", ["Amusementpark", "Teapot", "Sentiment Negative"]),
    string_col("SectionKeys", ["Interesting Colors", "Interesting Colors", "Other Colors"]),
    string_col("SectionLabels", ["Favorites", "Favorites", "Other"]),
    string_col("SectionDescriptions", ["Favorite colors", "Favorite colors", "Other colors"]),
    string_col("SectionIcons", ["Folder", "Folder", "Not Found"])
])

@ui.component
def ui_list_view():
    colors, set_colors = ui.use_state(["salmon", "lemonchiffon"])

    print(f"Colors are now {colors}")

    # this will create a controlled list_view with color_table
    return ui.list_view(
        color_table,
        key_column="Keys",
        label_column="Labels",
        description_column="Descriptions",
        icon_column="Icons",
        selected_keys=colors,
        on_change=set_colors
    )

my_list_view = ui_list_view()
```

**Comment by bmingles** [comment]
@mofojed I think the issue with selection not working is you have to set `selection_mode="multiple"`

**Comment by bmingles** [comment]
@mofojed I fixed the default selection_mode to be "MULTIPLE" and added handling for uppercase strings. Things should work as expected

File: plugins/ui/docs/README.md [hunk 1]
```diff
@@ -240,6 +240,91 @@ my_picker = ui_picker()
...
+        text,
+        direction="column",
+        margin=10,
+        gap=10,
+        # necessary to avoid overflowing container height
+        min_height=0,
+    )
+
+
+lv_table = ui_list_view_table()
```
Comment (mofojed) [inline plugins/ui/docs/README.md:325]: I got an error while running this second example, and the widget disappeared. Error in the logs:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'key')
    at $4e8b0456ef72939f$export$9610e69494fadfd2 (useGridListItem.ts:75:15)
    at $d7c07ca2efc5ba02$export$c6bde0c04b033c0e (ListViewItem.tsx:72:7)
    at renderWithHooks (react-dom.development.js:14985:18)
    at mountIndeterminateComponent (react-dom.development.js:17811:13)
    at beginWork (react-dom.development.js:19049:16)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:3945:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994:16)
    at invokeGuardedCallback (react-dom.development.js:4056:31)
    at beginWork$1 (react-dom.development.js:23964:7)
    at performUnitOfWork (react-dom.development.js:22776:12)
    at workLoopSync (react-dom.development.js:22707:5)
    at renderRootSync (react-dom.development.js:22670:7)
    at performSyncWorkOnRoot (react-dom.development.js:22293:18)
    at react-dom.development.js:11327:26
    at unstable_runWithPriority (scheduler.development.js:468:12)
    at runWithPriority$1 (react-dom.development.js:11276:10)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11322:9)
    at flushSyncCallbackQueue (react-dom.development.js:11309:3)
    at flushSync (react-dom.development.js:22467:7)
    at ScrollView.tsx:79:5
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:3945:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994:16)
    at invokeGuardedCallback (react-dom.development.js:4056:31)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:4070:25)
    at executeDispatch (react-dom.development.js:8243:3)
    at processDispatchQueueItemsInOrder (react-dom.development.js:8275:7)
    at processDispatchQueue (react-dom.development.js:8288:5)
    at dispatchEventsForPlugins (react-dom.development.js:8299:3)
    at react-dom.development.js:8508:12
    at batchedEventUpdates$1 (react-dom.development.js:22396:12)
    at batchedEventUpdates (react-dom.development.js:3745:12)
    at dispatchEventForPluginEventSystem (react-dom.development.js:8507:3)
    at attemptToDispatchEvent (react-dom.development.js:6005:3)
    at dispatchEvent (react-dom.development.js:5924:19)
    at unstable_runWithPriority (scheduler.development.js:468:12)
    at dispatchUserBlockingUpdate (react-dom.development.js:5894:5)
```

Using the latest `main` (`c3ea8670`)

File: plugins/ui/docs/README.md [hunk 2]
```diff
@@ -240,6 +240,91 @@ my_picker = ui_picker()
 
 ![Use a picker to select from a list of items](_assets/picker.png)
 
+## ListView (string values)
+A list view that can be used to create a list of selectable items. Here's a basic example for selecting from a list of string values and displaying the selected key in a text field.
+
+```python
+import deephaven.ui as ui
```
Comment (mofojed) [inline plugins/ui/docs/README.md:247]: Should be consistent with the rest of the examples, `from deephaven import ui`
```suggestion
from deephaven import ui
```

File: plugins/ui/docs/README.md [hunk 3]
```diff
@@ -240,6 +240,91 @@ my_picker = ui_picker()
...
+    return ui.grid(text, lv, lv2, columns="repeat(2, 1fr)")
+
+
+lv = ui_list_view()
+```
+
+## ListView (table)
+```python
+import deephaven.ui as ui
+from deephaven import time_table
```
Comment (mofojed) [inline plugins/ui/docs/README.md:283]:
```suggestion
from deephaven import time_table, ui
```

File: plugins/ui/docs/README.md [hunk 4]
```diff
@@ -240,6 +240,91 @@ my_picker = ui_picker()
...
+        ui.item("Item 1", key="Text 1"),
+        ui.item("Item 2", key="Text 2"),
+        ui.item("Item 3", key="Text 3"),
+        on_change=set_value,
+        selected_keys=value,
+    )
+
+    text = ui.text("Selection: " + ", ".join(map(str, value)), grid_column="span 2")
+
+    return ui.grid(text, lv, lv2, columns="repeat(2, 1fr)")
```
Comment (mofojed) [inline plugins/ui/docs/README.md:274]: This example doesn't appear correctly for me as is. Currently the list views do not appear at all (the list-view-wrapper has a height of zero):
image

I'd just keep it simple and just return them in a column/default layout, no real need for a grid here after Don's changes, e.g.
```suggestion
    text = ui.text("Selection: " + ", ".join(map(str, value)))

    return text, lv, lv2
```

Or if we still want the list views side by side:
```
    text = ui.text("Selection: " + ", ".join(map(str, value)))

    return text, ui.flex(lv, lv2, flex_grow=1)
```

I'm guessing Don's flex/CSS changes broke this for you, but we'll need to update the example.

File: plugins/ui/docs/README.md [hunk 5]
```diff
@@ -240,6 +240,91 @@ my_picker = ui_picker()
...
+def ui_list_view():
+    value, set_value = ui.use_state(["Text 2"])
+
+    # list_view with text children
+    lv = ui.list_view(
+        "Text 1",
+        "Text 2",
+        "Text 3",
+        on_change=set_value,
+        selected_keys=value,
```
Comment (mofojed) [inline plugins/ui/docs/README.md:260]: Selection isn't working for me at all right now. The second item should be selected initially, and it should update when things are checked off.

Looks like the props aren't actually passed through in @deephaven/components/ListView: https://github.com/deephaven/web-client-ui/blob/a283e13fafe1ecb156985fab00ba15344f180ff4/packages/components/src/spectrum/listView/ListView.tsx#L55

Unsure why that didn't get flagged as an unused variable ... 

Also getting a ton of warnings in the logs whenever I change selection, we should squash these warnings or provide default values so Spectrum doesn't complain: 
```
An aria-label or aria-labelledby prop is required for accessibility. 
    at $f85fb77f9d4cbc6c$var$ListView (
```

File: plugins/ui/docs/README.md (hunk 1 above)
Comment (bmingles) [reply plugins/ui/docs/README.md:325]: This looks like the Spectrum scroll bug:
https://github.com/adobe/react-spectrum/issues/6181

File: plugins/ui/docs/README.md (hunk 5 above)
Comment (bmingles) [reply plugins/ui/docs/README.md:261]: Selection should be fixed by https://github.com/deephaven/web-client-ui/pull/1986 .

I added the aria-label props to examples.

**Comment by bmingles** [comment]
@mofojed Matt fixed the underlying e2e kerning issue, so this PR should be unblocked now

**Review by mofojed — APPROVED** [review]
Update the PR description before merging.

**Pull request merged**